diesel/src/query_dsl/mod.rs

```diff
@@ -199,6 +199,15 @@ pub trait QueryDsl: Sized {
     /// column. `title.nullable()`, `lower(title).nullable()`, and `(id,
     /// title).nullable()` would all be valid.
     ///
+    /// Note that in order to use this method with a method like [`.inner_join`] or [`.filter`],
+    /// you will need to use it before calling [`.select`] (See examples below).
+    /// This is because you can only access columns from tables
+    /// that appear in your query before that function call.
+    ///
+    /// [`.inner_join`]: QueryDsl::inner_join()
+    /// [`.filter`]: QueryDsl::filter()
+    /// [`.select`]: QueryDsl::select()
+    ///
```

Comment:
This should only mention `inner_join`/`left_join`, but **not** `filter. The ordering of `.select` and `filter` calls does not matter.

I also would like to suggest a slightly different wording here:
```suggestion
    /// In order to use this method with columns from different tables 
    /// a method like [`.inner_join`] or [`.left_join`] needs to be called before 
    /// calling [`.select`] (See examples below).
    /// This is because you can only access columns from tables
    /// that appear in your query before that function call.
    ///
    /// [`.inner_join`]: QueryDsl::inner_join()
    /// [`.left_join`]: QueryDsl::left_join()
    /// [`.select`]: QueryDsl::select()
    ///
```